<?xml version="1.0"?>
<robot name="my_maniplator">
  <link name="base_link">
  </link>

  <joint name="my_joint" type="fixed">
    <parent link="base_link"/>
    <child link="link_0"/>
  </joint>

  <link name="link_0">
    <visual>
      <origin xyz="0 0 0.2" rpy="0 0 0"/>
      <geometry>
        <cylinder length="0.4" radius="0.5"/>
      </geometry>
    </visual>
  </link>

  <joint name="joint_0" type="revolute">
    <origin xyz="0 0 0.4" rpy="0 0 3.1416"/>
    <parent link="link_0"/>
    <child link="link_1"/>
    <axis xyz="0 0 1"/>
    <limit effort="30" velocity="1.0" lower="-1" upper="1" />
  </joint>

  <link name="link_1">
    <visual>
      <origin xyz="0 0 0.3" rpy="0 0 0"/>
      <geometry>
        <box size="0.4 0.4 0.6"/>
      </geometry>
    </visual>
  </link>

  <joint name="joint_1" type="revolute">
    <origin xyz="0 0 0.6" rpy="0 0 3.1416"/>
    <parent link="link_1"/>
    <child link="link_2"/>
    <axis xyz="1 0 0"/>
    <limit effort="30" velocity="1.0" lower="-1" upper="1" />
  </joint>

  <link name="link_2">
    <visual>
      <origin xyz="0 0 0.3" rpy="0 0 0"/>
      <geometry>
        <box size="0.3 0.3 0.6"/>
      </geometry>
    </visual>
  </link>

  <joint name="joint_2" type="revolute">
    <origin xyz="0 0 0.6" rpy="0 0 3.1416"/>
    <parent link="link_2"/>
    <child link="link_3"/>
    <axis xyz="1 0 0"/>
    <limit effort="30" velocity="1.0" lower="-1" upper="1" />
  </joint>

  <link name="link_3">
    <visual>
      <origin xyz="0 0 0.45" rpy="0 0 0"/>
      <geometry>
        <box size="0.25 0.25 0.9"/>
      </geometry>
    </visual>
  </link>

</robot>
